linker/shared/src/main/scala/org/scalajs/linker/Linker.scala

```diff
@@ -20,7 +20,7 @@ import org.scalajs.linker.irio._
  *  module initializers, link them together, and write the output to a writable
  *  .js file.
  */
-trait GenLinker {
+abstract class Linker {
```

Comment:
The constructor here should be private so that we can add abstract methods in minor versions.

Reply:
Right, that's indeed what we designed in #2998. But eventually we should be allowed to open this up. We might want to allow users to write their own `Linker` (which would probably wrap an existing standard linker, not reimplement everything). For example that's what I had done in https://github.com/sjrd/scalajs-reflect.